core_java/jdbc/src/main/java/ca/jrvs/apps/jdbc/CustomerDAO.java

```diff
@@ -0,0 +1,135 @@
+package ca.jrvs.apps.jdbc;
+
+import ca.jrvs.apps.jdbc.util.DataAccessObject;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CustomerDAO extends DataAccessObject<Customer> {
+
+    final Logger logger = LoggerFactory.getLogger(CustomerDAO.class);
+
+    private static final String INSERT = "INSERT INTO customer (first_name, last_name, " +
+            "email, phone, address, city, state, zipcode) VALUES (?, ?, ?, ?, ?, ?, ?, ?)";
+
+    private static final String GET_ONE = "SELECT customer_id, first_name, last_name, email, phone, address, city, " +
+            "state, zipcode FROM customer Where customer_id = ?";
+
+    private static final String UPDATE = "UPDATE customer SET first_name = ?, last_name=?, email = ?, phone = ?, " +
+            "address = ?, city = ?, state = ?, zipcode = ? WHERE customer_id = ?";
+
+    private static final String DELETE = "DELETE FROM customer WHERE customer_id = ?";
+
+    public CustomerDAO(Connection connection) {
+        super(connection);
+    }
+
+    @Override
+    public Customer findById(long id) {
+        Customer customer = new Customer();
+        try (PreparedStatement statement = this.connection.prepareStatement(GET_ONE);) {
+            statement.setLong(1, id);
+            ResultSet rs = statement.executeQuery();
+            while (rs.next()) {
+                customer = getCustomer(rs);
+            }
+        } catch (SQLException e) {
+            this.logger.error(e.getMessage(), e);
+            throw new RuntimeException(e);
+        }
+        return customer;
+    }
+
+    @Override
+    public List<Customer> findAll() {
+        return null;
+    }
+
+    @Override
+    public Customer update(Customer dto) {
+        Customer customer;
+        try {
+            this.connection.setAutoCommit(false);
+        } catch (SQLException e) {
+            this.logger.error(e.getMessage(), e);
```

Comment:
you don't need to use `this` keyword unless there is a naming conflict. (e.g. `this.name = name`)